field/TinyLang/Field/Generator.hs

```diff
@@ -269,6 +201,51 @@ arbitraryUnOpRing = elements [Neg]
 arbitraryBinOpRing :: MonadGen m => m (BinOp f (AField f) (AField f) (AField f))
 arbitraryBinOpRing = elements [Add, Sub, Mul]
 
+-- | Generate an expression of a particular type from a collection of variables
+-- with the number of nodes (approximately) bounded by 'size'.
+boundedArbitraryExpr
+    :: forall f a m. (Field f, Arbitrary f, KnownUni f a, MonadGen m, MonadSupply m)
+    => Vars f -> Int -> m (Expr f a)
+boundedArbitraryExpr _    size | size <= 1 = EVal <$> arbitraryM
```

Comment:
I think we should probably add EVar here as well.  This line is supposed to add a single leaf when we reach the limit for the number of nodes that are required.  I don't see any reason why it should just be a value and not a variable.  This is probably just a mistake that's been carried over from the original version.

Reply:
Done. I only changed `boundedArbitraryExpr` and not `boundedArbitraryExprI`, because there we do not generate variables anyway.